index.js

```diff
@@ -189,7 +189,7 @@ S3Storage.prototype._handleFile = function (req, file, cb) {
       params.ContentDisposition = opts.contentDisposition
     }
 
-    if (this.throwMimeTypeConflictErrors && (file.mimetype !== opts.contentType)) {
+    if (this.throwMimeTypeConflictErrors && (opts.contentType !== 'application/octet-stream') && (file.mimetype !== opts.contentType)) {
```

Comment:
it may be better to allow a user-defined function `mimeTypeFilter` here instead of simple boolean. example:
```
if (typeof this.mimeTypeFilter === 'function' && !this.mimeTypeFilter(opts.contentType, file.mimeType) {
```

this is more flexible for consumers and it makes this code easier to maintain in my opinion.

i will update this pull request with those changes. thanks!